models/issue.go

```diff
@@ -440,7 +440,7 @@ func (issue *Issue) HashTag() string {
 
 // IsPoster returns true if given user by ID is the poster.
 func (issue *Issue) IsPoster(uid int64) bool {
-	return issue.PosterID == uid
+	return issue.OriginalAuthorID == 0 && issue.PosterID == uid
```

Comment:
Just to confirm from what I have understand when migrating if the user exist OriginalAuthorID  is not and set and issue.PosterID is not the doer (like it is by default) but found user ?

Reply:
If OriginalAuthorID > 0, PosterID means migrator but not poster id. `OriginalAuthorID` is an external system id not gitea user id.